library/general/requiredlabels/template.yaml

```diff
@@ -38,31 +38,41 @@ spec:
                       the regular expression.
   targets:
     - target: admission.k8s.gatekeeper.sh
-      rego: |
-        package k8srequiredlabels
+      code:
+      - engine: K8sNativeValidation
+        source:
+          validations:
+          - expression: '(has(request.operation) && request.operation == "DELETE") ||(has(object.metadata) && variables.params.labels.all(entry, has(object.metadata.labels) && entry.key in object.metadata.labels))'
```

Comment:
We do not need to handle DELETE as a special case. CEL-VAP sees the same input objects that Rego does. We should do a straight re-implementation of the Rego logic (which does not special-case delete).

In general, mirroring the Rego code's logic exactly is the correct move, otherwise there will be edge cases where the two implementations behave differently.

Reply:
hmm, recall when we first integrated vap, I had to explicitly check `oldObject` instead of `object` because on DELETE, the expression `object.metadata` was failing so it was always blocking on DELETE.

Reply:
I forgot we changed this part of  the generated vap resource so DELETE is ignored.
```yaml
operations:
   - CREATE
   - UPDATE
   - ```